File: mySolutions/14. Longest Common Prefix/14. Longest Common Prefix.java
class Solution {
    public String longestCommonPrefix(String[] strs) {
        
        if (strs.length == 0) {
            return "";
        }
        
        String longest = strs[0];
        
        int i = 0;
        
        while (i < strs.length) {
            while (strs[i].indexOf(longest) != 0) {
                    longest = longest.substring(0, longest.length() - 1);
            }
            i++;
        }
        
        return longest;
    }
}

File: mySolutions/14. Longest Common Prefix/14. Longest Common Prefix.py
class Solution:
    def longestCommonPrefix(self, strs: List[str]) -> str:
      if len(strs) == 0:
        return ""
      
      start = 0
      shortest = None
      
      for str in strs:
        if shortest == None or len(shortest) > len(str):
          shortest = str
      for end in range(0, len(shortest)):
        
        for str in strs:
          if str[end] != strs[0][end]:
            return strs[0][0: end]
          
      return strs[0][0: len(shortest)]